### DISCUSSION TITLE
FR: add switch domain to magic areas

### DISCUSSION THREAD

**Discussion by xBourner** [jseidl/magic-areas#403] [Ideas / Feature Requests]
Could you add switch entities to magic areas like all other entities?
So for each room and global (meta areas)

Would like to see some diffferences based on device_class attribute. Like in this script:

                {%- set areas = states.switch | selectattr('attributes.device_class','eq','outlet') 
                | map(attribute='entity_id') 
                | unique | reject('none') | list -%} 
                {{areas}}

**Comment by jseidl** [comment]
You can create a binary_sensor of device class presence template sensor based-off that switch or use the "Change type of switch" helper. Magic Areas can't automatically consider `switch` because they will not be always related to presence. Not all outlets are lights or presence sensors.
